Mage/src/main/java/mage/abilities/SpellAbility.java

```diff
@@ -141,12 +143,26 @@ public ActivationStatus canActivate(UUID playerId, Game game) {
                 }
             }
 
+            // TODO: this check may not be required, but removing it require more investigation.
+            //       As of now it is only a way for One with the Multiverse to work.
+            if (!approvingObjects.isEmpty()) {
+                Card card = game.getCard(sourceId);
+                Zone zone = game.getState().getZone(sourceId);
+                if(card != null && card.isOwnedBy(playerId) && Zone.HAND.match(zone)) {
+                    // Regular casting, to be an alternative to the AsThoughEffectType.PLAY_FROM_NOT_OWN_HAND_ZONE from hand (e.g. One with the Multiverse):
+                    approvingObjects.add(new ApprovingObject(this, game));
+                }
+            }
+
             // no mana restrict
             // Alternate spell abilities (Flashback, Overload) can't be cast with no mana to pay option
             if (getSpellAbilityType() == SpellAbilityType.BASE_ALTERNATE) {
                 Player player = game.getPlayer(playerId);
                 if (player != null
-                        && player.getCastSourceIdWithAlternateMana().contains(getSourceId())) {
+                        && player.getCastSourceIdWithAlternateMana()
+                                .getOrDefault(getSourceId(), new HashSet<>())
```

Comment:
this is another example where Collections.emptySet() would make more sense than initializing an empty HashSet just to check what it contains